### PR TITLE
[IMP] sale_crm: propagate fields on quotation creation from stat button

### PR DESCRIPTION
**Pull request by Odoonan**
When using the 'new quotation' button on the opportunity view, the tags of the
opportunity are propagated on the quotation. However, this is not the case when
using the stat button 'quotations' and creating a quotation from there.

This commits adds the tags in context to propagate them in that case as well,
as well as the other fields propagated on the 'new quotation' button:
medium, campaign, ...

This means both behaviours now have the same context, except for the search
default values.

--- LINKS ---
Task Id - 2679488

### CODE REVIEW

File: addons/sale_crm/models/crm_lead.py [hunk 1]
```diff
@@ -87,6 +73,19 @@ def action_view_sale_order(self):
             action['res_id'] = orders.id
         return action
 
+    def _get_opportunity_quotation_context(self):
```
Comment (awa-odoo) [inline addons/sale_crm/models/crm_lead.py:76]: Let's go with "_prepare_opportunity_quotation_context".

We usually use the "prepare" keyword for methods that prepare a dict of values.

Side note: Could add a bit of docstring to help.
Something like:

""" Prepares the context for a new quotation (sale.order) by sharing the values of common fields. """"

File: addons/sale_crm/models/crm_lead.py [hunk 2]
```diff
@@ -87,6 +73,19 @@ def action_view_sale_order(self):
...
+        return {
```
Comment (awa-odoo) [inline addons/sale_crm/models/crm_lead.py:77]: Could put a `self.ensure_one()` line before the return to make sure.

File: addons/sale_crm/models/crm_lead.py [hunk 3]
```diff
@@ -60,12 +50,8 @@ def action_new_quotation(self):
...
     def action_view_sale_quotation(self):
         action = self.env["ir.actions.actions"]._for_xml_id("sale.action_quotations_with_onboarding")
-        action['context'] = {
-            'search_default_draft': 1,
-            'search_default_partner_id': self.partner_id.id,
-            'default_partner_id': self.partner_id.id,
-            'default_opportunity_id': self.id
-        }
+        action['context'] = self._get_opportunity_quotation_context()
+        action['context'].update({'search_default_draft': 1})
```
Comment (awa-odoo) [inline addons/sale_crm/models/crm_lead.py:54]: When it's a single value you can use the key directly.

`action['context']['search_default_draft'] = 1`

(Same remark for here above)

**Review by awa-odoo — COMMENTED** [review]
Hello @Odoonan 

Overall looks good!
I added a few polishing comments.

Side note: the task link at the end should be "Task-123456".
It's "important" because if your task ID matches the end of a commit hash, github will try to create a link (and the link will be wrong of course).
But if the number is attached to another word, it will not attempt to do it.
Not a big deal but it's a good habit to have :slightly_smiling_face: 

Cheers!

File: addons/sale_crm/models/crm_lead.py [hunk 4]
```diff
@@ -87,6 +73,21 @@ def action_view_sale_order(self):
             action['res_id'] = orders.id
         return action
 
+    def _prepare_opportunity_quotation_context(self):
+        """ Prepares the context for a new quotation (sale.order) by sharing the values of common fields """
+        self.ensure_one()
```
Comment (tde-banana-odoo) [inline addons/sale_crm/models/crm_lead.py:78]: While we are at it, shoudln't we do the user / team thingy done in ``action_new_quotation`` also ?
```
        if self.team_id:
            action['context']['default_team_id'] = self.team_id.id,
        if self.user_id:
            action['context']['default_user_id'] = self.user_id.id
```

**Review by tde-banana-odoo — COMMENTED** [review]
Riviou !

**Comment by Odoonan** [comment]
@tde-banana-odoo  Good point! :) All done here!

ps: I cannot include default team and user directly in the returned dictionary  (e.g writing something like `'default_team_id' : self.team_id.id if self.team_id else False` ). because it would erase the default default behaviour, using False values instead of the connected user, for instance. The previous behavior is maintained by using the same syntax in the new method.

Cheers!

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo r+

**Pull request closed without merging**